gpii/node_modules/lifecycleManager/test/js/LifecycleManagerTests.js

```diff
@@ -64,16 +64,62 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
         return togo;
     };
 
-    gpii.tests.lifecycleManager.buildSettingsHandlersEntry = function (settings) {
-        return {
+    /**
+     * Used by a bunch of the tests to avoid having to write out the almost-identical full set of lifecycle
+     * instructions for each test. Instead one can have a "base set" of lifecycle instructions
+     * that can be shared across multiple tests/multiple phases of a single test, and then use
+     * this function to extend base set to match the specific test/test phase.
+     *
+     * The function takes an unlimited number of arguments. The first argument is mandatory and must
+     * be a set of lifecycle instructions entries keyed by solution ids. While the system wont explode
+     * if entries for multiple solutions are provided, only the "first" solution entry will be
+     * extended. The remaining arguments will be fluid.extended onto the first argument.
+     *
+     * @currentInstructions {Object} a set of lifecycle instructions keyed by a solution id. For example,
+     *    this could be the output of the buildLifecycleInstructions function.
+     * @arguments {..Object..} The remaining arguments should be valid lifecycleInstructions that will
+     *    be extended onto the original
+     * @return {Object} The original lifecycle instructions, but with the remaining arguments fluid.extend'ed
+     *    onto the first solution entry in the currentInstructions
+     */
+    gpii.tests.lifecycleManager.extendLifecycleInstructions = function (currentInstructions) {
+        var togo = fluid.copy(currentInstructions);
+        var solId = Object.keys(currentInstructions)[0];
```

Comment:
I think it would well be worth having a check that the length of Object.keys is exactly 1 and if it isn't, exploding